gradebook/src/GradeBook/book.cs

```diff
@@ -0,0 +1,71 @@
+using System;
+using System.Collections.Generic;
+
+/*
+The class Book packages the blueprint book provides ways to:
+1 store the grades. 
+2 add to the given gradebook obj
+3 welcomes the user from args
+4 Assign name to the gradebook
+ */
+
+namespace GradeBook
+{
+    public class book
+    {
+        List<double> grades;
+        string name;
+        double highGrade = double.MinValue;
+        double lowGrade = double.MaxValue;
+        double avg = 0.0;
+
+        //Constructor
+        public book(string name, string[] args) 
+        {
+            grades = new List<double>() {1,22,56,33.2,45,30,32};
+            this.name = name;
+            welcomeUser(args);
+        }
+
+        public void calculateStats()
+        {
+            foreach(double gd in this.grades)
+            {
+                avg += gd;
+                highGrade = Math.Max(highGrade,gd);
+                lowGrade = Math.Min(lowGrade, gd);
+            }
+            avg /= this.grades.Count;
+        }
+
+        // show the current statistics
+        public void showStatistics() 
+        {
+            calculateStats();
+            Console.WriteLine($"Min grade - {lowGrade}\nMax Grade- {highGrade}\nAvg Grade - {avg}");
+        }
+
+        // Welcomes the user
+        private void welcomeUser(string[] args)
+        {
+            string user = "User";
+            if(args.Length>0)
+            {
+                user = args[0];
+            }
+            Console.WriteLine($"Welcome, {user}");
+        }
+
+        // Adds grade to the class member
+        public void AddGrade(double grade) 
+        {
+            this.grades.Add(grade);
+        }
+
+        // getter for grades member
+        public List<double> getGrade()
```

Comment:
Don't think the grades field should be exposed.  If it has to be, look into `ReadOnlyList` or `IEnumerable` - other classes shouldn't be able to change your private data under you.

Reply:
That makes sense. I can just let external world to use it via a getter function.

Reply:
Having read-only access can make sense, but since you're passing back a full `List`, the caller can also change the data without you knowing about it.  In situations like this you would usually want to use an `IEnumerable`, if iterating through the grades is the only thing required, or a `ReadOnlyList`, if you need to have random access to the grades.  .NET also prefers [properties](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/properties) over getters and setters.

Reply:
added get property and Read only list type case while returning the private member. 
This was a nice one. I got to read and implement a couple of things due to this comment of yours. Thanks 👍